client/src/partials/cash/extra_payment/extra_payment.js

```diff
@@ -49,66 +46,125 @@ angular.module('bhima.controllers')
         tables : {
-          'account' :{
-            columns : ['id', 'account_txt', 'account_number']
+          'currency' : {
+            columns : ['id', 'name', 'symbol']
           }
-        },
-        where : ['account.classe=4']
+        }
       }
     };
 
     cache.fetch('account').then(getAccount);
 
-    appstate.register('project', function (project) {
-      $scope.project =  project;
-      validate.process(dependencies)
-      .then(init)
-      .then(getSale)
-      .catch(function (err) {
-        messenger.danger(err);
-      });
-    });
+    $scope.formatPatient = function (patient) {
+      return patient ? [patient.first_name, patient.last_name].join(' ') : '';
+    };
 
-    function getSale(data) {
-      // FIXME : utilisation abusive des boucles
-      // l'algorithme marche mais n'est pas optimal
-      $scope.saleData = [];
-      data.forEach(function (sale) {
-        $http.get('/ledgers/debitor/'+sale.debitor_uuid)
-        .success(function (rows) {
-          var items = rows.filter(function (row) {
-            return row.balance > 0;
-          });
-          items.forEach(function (row) {
-            row.debitor_first = sale.first_name;
-            row.debitor_middle = sale.middle_name;
-            row.debitor_last = sale.last_name;
-            row.transaction = row.abbr + row.reference;
-            row.cost = row.balance;
-            row.currency = 2; // values are always in $ by default
-            row.currency_id = 2; // values are always in $ by default
+    $scope.formatAccount = function (ac) {
+      if(ac){return ac.account_number + ' - ' + ac.account_txt;}
+    };
+
+    function processModels(models) {
+      angular.extend(session, models);
+      session.date = new Date();
+    }
+
+    function handleErrors(err) {
+      messenger.danger('An error occured:' + JSON.stringify(err));
+    }
+
+    function getAccount(ac) {
+      if (!ac) { return; }
+       session.configured = true;
+       session.ac = ac;
+       session.complete = true;
+    }
+
+    function search() {
+      $scope.state = 'generate';
+      session.patient = session.selected;
+      session.configured = true;
+
+      var id = session.patient.debitor_uuid,
+        account_id = session.patient.account_id;
+
+      dependencies.sales = {
+        query : {
+          tables : {
+            'sale' : {
+              columns : ['uuid', 'debitor_uuid']
+            },
+            'patient' : {
+              columns : ['first_name', 'middle_name', 'last_name']
+            }
+          },
+          join : ['sale.debitor_uuid=patient.debitor_uuid'],
+          where : ['patient.debitor_uuid=' + id]
+        }
+      };
+      validate.process(dependencies, ['sales'])
+      .then(function (model) {
+        var sales = model.sales.data;
+        $scope.saleData = [];
+        sales.forEach(function (sale) {
+          $http.get('/ledgers/debitor/'+sale.debitor_uuid)
+          .success(function (rows) {
```

Comment:
This generates an $http request for each debtor.  Is it possible to change the code so that we only generate one?